addons/hr_recruitment/models/hr_recruitment.py

```diff
@@ -282,6 +283,27 @@ def _onchange_stage_id_internal(self, stage_id):
             return {'value': {'date_closed': fields.datetime.now()}}
         return {'value': {'date_closed': False}}
 
+    @api.depends('user_id')
+    def _compute_date_open(self):
+        if self.user_id:
+            self.write({'date_open': fields.Datetime.now()})
+
+    @api.depends('stage_id')
+    def _compute_kanban_state(self):
+        if self.stage_id:
+            self.write({'kanban_state': 'normal'})
```

Comment:
Before the kanban_state was reset to normal on stage_id update when no new kanban_state was supplied. It is now written every time.